src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double CAPACITY_INCREASE_MODIFICATOR = 1.5;
+    private Object[] defaultList;
+    private int size;
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity <= 0) {
+            throw new ArrayListIndexOutOfBoundsException("Initial capacity must be > 0");
+        }
+        defaultList = new Object[initialCapacity];
+    }
+
+    public ArrayList() {
+        this(DEFAULT_SIZE);
+    }
+
     @Override
     public void add(T value) {
-
+        increaseCapacity();
+        defaultList[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
```

Comment:
It is possible to unite this code with 'checkIndex' method to avoid duplication, maybe 'checkindex' should take an additional argument or be overloaded